hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/SimpleForeignKeyDescriptor.java

```diff
@@ -184,7 +180,6 @@ public Predicate generateJoinPredicate(
 			);
 		}
 		else {
-			assert rhsTableExpression.equals( keyColumnContainingTable );
```

Comment:
One of my current ongoing entity graph testing case failed because of the above statement. Glad it has been removed.